components/asset.rst

```diff
@@ -0,0 +1,251 @@
+
+    echo $package->getUrl('/image.png');
+    // result: /image.png?v1
+
+In case you want to modify the version format, pass a sprintf-compatible format
+string as the second argument of the ``StaticVersionStrategy`` constructor::
+
+    // put the 'version' word before the version value
+    $package = new Package(new StaticVersionStrategy('v1', '%s?version=%s'));
+
+    echo $package->getUrl('/image.png');
+    // result: /image.png?version=v1
+
+    // put the asset version before its path
+    $package = new Package(new StaticVersionStrategy('v1', '%2$s/%1$s'));
+
+    echo $package->getUrl('/image.png');
+    // result: /v1/image.png
+
+Custom Version Strategies
+.........................
+
+Use the ``VersionStrategyInterface`` to define your own version strategy. For
+example, you could define a versioning where the current date is appended to
+bust the cache every day::
+
+    use Symfony\Component\Asset\VersionStrategy\VersionStrategyInterface;
+
+    class DateVersionStrategy implements VersionStrategyInterface
+    {
+        private $version;
+
+        public function __construct()
+        {
+            $this->version = date('Ymd');
+        }
+
+        public function getVersion($path)
+        {
+            return $this->version;
+        }
+
+        public function applyVersion($path)
+        {
+            return sprintf('%s?v=%s', $path, $this->getVersion($path));
+        }
+    }
+
+Grouped Assets
+~~~~~~~~~~~~~~
+
+It's common for applications to store their assets in a common path. If that's
+your case, replace the default ``Package`` class by ``PathPackage`` to avoid
+repeating the same path time and again::
+
+    use Symfony\Component\Asset\PathPackage;
+
+    $package = new PathPackage('/static/images', new StaticVersionStrategy('v1'));
+
+    echo $package->getUrl('/logo.png');
+    // result: /static/images/logo.png?v1
+
+Request Context Aware Assets
+............................
+
+If you are also using the HttpFoundation component in your project, for example
+in a Symfony application, the ``PathPackage`` class can take into account the
+context of the current request::
+
+    use Symfony\Component\Asset\PathPackage;
+    use Symfony\Component\Asset\Context\RequestStackContext;
+
+    $package = new PathPackage('/static/images', new StaticVersionStrategy('v1'));
+    $package->setContext(new RequestStackContext($requestStack));
+
+    echo $package->getUrl('/logo.png');
+    // result: /somewhere/static/images/logo.png?v1
+
+When the request context is set, in addition to the configured base path,
+``PathPackage`` also prepends the current request base URL (``/somewhere/`` in
+this example) to assets. This allows your website to be hosted anywhere under
+the web server root directory.
+
+Absolute Assets and CDNs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Applications that host their assets on different domains and CDNs (*Content
+Delivery Networks*) should use instead the ``UrlPackage`` class to generate
```

Comment:
"should use instead the `UrlPackage` class" -> "should use the `UrlPackage` class instead"